src/main/java/com/microsoft/lst_bench/common/DependentTaskExecutor.java

```diff
@@ -0,0 +1,119 @@
+ * first statement is stored in an intermediate object that can be specific to the connection. The
+ * expected object for a JDBC connection is of type List<Map<String, Object>>, handling of other
+ * objects would need to be added to the if-clause that checks the instance of the object.
+ */
+public class DependentTaskExecutor extends TaskExecutor {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(DependentTaskExecutor.class);
+  // TODO: Add functionality to use custom replacement markers.
+  private static final String DEFAULT_REPLACEMENT_MARKER = "dependent_replace";
+
+  private final Integer dependentBatchSize;
+
+  public DependentTaskExecutor(
+      SQLTelemetryRegistry telemetryRegistry,
+      String experimentStartTime,
+      Integer dependentBatchSize) {
+    super(telemetryRegistry, experimentStartTime);
+    // Set to a default of '1' if not otherwise specified.
+    if (dependentBatchSize == null) {
+      this.dependentBatchSize = 1;
+    } else {
+      this.dependentBatchSize = dependentBatchSize;
+    }
+  }
+
+  @Override
+  public void executeTask(Connection connection, TaskExec task, Map<String, Object> values)
+      throws ClientException {
+    for (FileExec file : task.getFiles()) {
+      Instant fileStartTime = Instant.now();
+      try {
+        for (int i = 0; i < file.getStatements().size(); i += 2) {
+          StatementExec statement = file.getStatements().get(i);
+
+          // Execute first query that retrieves the iterable input for the second query.
+          Instant statementStartTime = Instant.now();
+          QueryResult queryResult =
+              connection.executeQuery(
+                  StringUtils.replaceParameters(statement, values).getStatement());
+          writeStatementEvent(statementStartTime, statement.getId(), Status.SUCCESS);
+
+          // Execute second query repeatedly with the parameters extracted from the first query.
+          if (queryResult != null) {
+            int size = queryResult.getValueListSize();
+            statement = file.getStatements().get(i + 1);
+            for (int j = 0; j < size; j += this.dependentBatchSize) {
+              int localMax =
+                  (j + this.dependentBatchSize) > size ? size : (j + this.dependentBatchSize);
+              Map<String, Object> localValues = new HashMap<>(values);
+              localValues.put(
+                  DEFAULT_REPLACEMENT_MARKER,
+                  this.createReplacementString(queryResult, j, localMax));
+
+              statementStartTime = Instant.now();
+              connection.execute(
+                  StringUtils.replaceParameters(statement, localValues).getStatement());
+              writeStatementEvent(statementStartTime, statement.getId(), Status.SUCCESS);
+            }
+          } else {
+            LOGGER.warn("QueryResult returned by first statement was empty, continuing execution.");
```

Comment:
Isn't this unexpected and should directly fail instead? Otherwise it could mask some errors inadvertently.

Reply:
The problem is the test which mocks the connection, in case of an error, the test would fail.